Cisco-AMP/amp4e_splunk_events_input
# Add Splunk 8 / Python 3 Support

With version 8.0, Splunk now supports Python 3. They also require that apps be compatible with Python 3 in order to be considered compatible with Splunk 8.0 (although ideally it is compatible with both Python 2 and 3). To ensure that our app is compatible for the most people, we will be maintaining compatibility with Python 2 for app versions targeting Splunk 8.0.

To do so, this PR will just **get the app working** with Splunk 8.0 and update Docker and Travis as necessary.
- [x] Update Dockerfile for compatibility with latest Splunk image
- [x] Run tests in both Python 2 and Python 3 on Travis
- [x] Tests are passing in Python 2
- [x] Tests are passing in Python 3

For upgrading to Python 3, the following changes were made:

- Moved function wrappers outside of classes
- Renamed `amp4e_events_input` package to `amp4e_events_input_lib` to avoid naming collision with `amp4e_events_input.py` when doing relative imports
- Change print statements to print function calls
- Change API error initialization so adding a message still works as before
- Update relative imports

Following this PR, we will need to do a full regression test of the app in both Python 2 and Python 3 to ensure compatibility.
